### DISCUSSION TITLE
About .db file

### DISCUSSION THREAD

**Discussion by callmehanu** [toeverything/AFFiNE#2558] [General]
Hi guys

First time I tried Affine, I noticed that all my data was stored  in a **.db file**.
After using Logseq and Obsidiam for months I supposed that all my page were contained over a exposed folder with "tons" of .txt or md files.

That allow me to switch my content (personal knowledge) beetween softwares and even use Logseq and Obsidian at the same time.

What are the **pros** of use **.db** estructure? Maybe I'm lost details but would really great to know a bit more this.

Sorry if is a dumb question! 🙏

**Comment by Yipei-Operation** [comment]
As for the design for this part, can check for: https://blocksuite.affine.pro/the-local-first-paradigm.html#local-first-in-affine (any related question welcome to comment to this discussion at any time!

**Comment by HeJiachen-PM** [comment]
Hello, long story short, we go with .db files for mainly two inevitable reasons:
1. There are many feature-rich blocks that cannot be expressed with markdown syntax. Especially when it comes to edgeless and databases. 
2. Features related to syncing or querying require an actual database, in our case, that is [Octobase](https://octobase.pro/). A .db file not only preserves documents but also relationships between docs, for the good of dual-link, collaboration and sync between devices. 

Deep inside, AFFiNE is taking advantage of CRDTs as our data type. We could "export" json or .md or XML for "general uses", but it makes more sense to have an "export plugin" rather than use them as the single source of truth of our data.
